### ISSUE TITLE
LSP doesn't recognize usage when alias is only used inside arguments declaration (schema integration)

### ISSUE THREAD

**Issue by lucasalencar** [snoe/clojure-lsp#122]
When working with functions declared using schema declaration, there is a warning indicating an unused namespace which is not true.

Example:

```clojure
(ns current-namespace
  (:require [schema.core :as s]
            [another-namespace :as an])) ;; Warning about unused namespace here

(s/defn function
  [argument :- an/Schema]
  (print argument))
```

What I suspect is that LSP doesn't understand that a schema declared in the arguments list is actually being used instead of just an argument declaration.

If I add the schema anywhere else, the warning goes away.

Example: 

```clojure
(ns current-namespace
  (:require [schema.core :as s]
            [another-namespace :as an])) 

(s/defn function :- an/Schema ;; Adding schema to the return, warning goes away
  [argument :- an/Schema]
  (print argument))
```

**Comment by snoe** [comment]
I think it's this line https://github.com/snoe/clojure-lsp/blob/master/src/clojure_lsp/parser.clj#L341-L342 skipping over both the :- and the potential symbol. Could probably check if it's a symbol and `add-reference`

**Comment by ericdallo** [comment]
I realized that `find-definition` is not working for the schemas on the arguments of a function, example: `find-definition` with the cursor on `an/Schema`.
@snoe do you think this fixing this issue should resolve this problem too?

**Comment by ericdallo** [comment]
It was [this change](https://github.com/snoe/clojure-lsp/commit/f38b40a55234bdbfa1bfb2c890112f578e9086f9#diff-f94fd8215111cb9c9ecc4498101f0586L799-L802) that caused this problem, rolling back it on config.edn fixed my problem and this issue!

**Comment by snoe** [comment]
right, that change allowed for multiple signatures.

**Comment by snoe** [comment]
Fixed by 2efa889

**Comment by ericdallo** [comment]
Thanks!
